safe_transaction_service/history/signals.py

```diff
@@ -112,6 +115,36 @@ def safe_master_copy_clear_cache(
     SafeMasterCopy.objects.get_version_for_address.cache_clear()
 
 
+def get_safe_addresses_from_instance(
+    instance: Union[
+        TokenTransfer,
+        InternalTx,
+        MultisigConfirmation,
+        MultisigTransaction,
+    ]
+) -> Tuple[Optional[ChecksumAddress], Optional[ChecksumAddress]]:
+
+    if isinstance(instance, TokenTransfer):
+        return (instance.to, instance._from)
+    elif isinstance(instance, MultisigTransaction):
+        return (instance.safe, None)
+    elif isinstance(instance, MultisigConfirmation) and instance.multisig_transaction:
+        return (instance.multisig_transaction.safe, None)
+    elif isinstance(instance, InternalTx):
+        return (instance.to, None)
+
+    return (None, None)
+
+
+def _clean_all_txs_cache(instance):
```

Comment:
Add docs. Also maybe another python file is better to add these helper methods, as we are merging a lot of things in the signals file. Maybe refactoring all this logic into a class (if possible) could make sense